hazelcast/src/main/java/com/hazelcast/topic/impl/TopicService.java

```diff
@@ -145,7 +147,14 @@ public void incrementReceivedMessages(String topicName) {
 
     public void publishEvent(String name, TopicEvent event) {
```

Comment:
What about this one? (by making `counter` an `AtomicInteger`)
- no need to create Long
- no need to find TopicConfig in every run

```
 public void publishEvent(String name, TopicEvent event, boolean multiThreadingEnabled) {
        Collection<EventRegistration> registrations = eventService.getRegistrations(TopicService.SERVICE_NAME, name);
        int orderKey = multiThreadingEnabled ? counter.incrementAndGet() : name.hashCode();
        eventService.publishEvent(TopicService.SERVICE_NAME, registrations, event, orderKey);

eventService.publishEvent(TopicService.SERVICE_NAME, registrations, event, multiThreadingEnabled ? orderKey)
    }
```